docs/test/test-guide.md

```diff
@@ -8,15 +8,129 @@ The minimum version of Go required is in the [https://github.com/redhat-develope
 
 #### Unit tests:
 
-Unit test does not require any cluster configuration, run `make test` to validate unit tests.
+Unit test does not require any cluster configuration, run `make test` to validate unit tests. Unit tests run on OpenShift CI before the code is merged. Unit test coverage is then reported to Codecov which provides coverage information on the pull requests.
 
-##### Prerequisites for OpenShift cluster:
+To run tests during the development cycle:
+```
+$ make test
+```
+To run specific tests, use one of the following methods:
+
+* Run all tests on a single package.
+    ```
+    # Eg: go test -v ./pkg/cmd/environment
+    $ go test -v <relative path of package>
+    ```
+
+* Run a single test on a single package.
+    ```
+    $ go test -v <relative path of package> -run <Testcase Name>
+    ```
+    
+* Run tests that match a pattern.
+    ```
+    $ go test -v <relative path of package> -run "Test<Regex pattern to match tests>"
+    ```
+For more information about test options, run the `go test --help` command and review the documentation.
+
+#### Prerequisites for OpenShift cluster:
 
 * A `crc` environment for 4.5+ local cluster:
 Follow [https://github.com/code-ready/crc#documentation](crc) installation guide.
 * Or a 4.5+ cluster hosted remotely
 
-NOTE: Make sure that `kam` and `oc` binaries are in `$PATH`. Use the cloned kam directory to launch tests on `4.5+` clusters. `4.5+` cluster needs to be configured before launching the tests against it. The files `kubeadmin-password` and `kubeconfig` which contain cluster login details should be present in the `auth` directory and it should reside in the same directory as `Makefile`. If it is not present in the auth directory, please create it. Then run `make prepare-test-cluster` to configure the `4.5+` cluster. `make prepare-test-cluster` comprises installation of sealed secrets, openshift pipelines, argocd operator and create sealed secrets instance.
+NOTE: Make sure that `kam`, `oc`, `gh` and `argocd` binaries are in `$PATH`. Use the cloned kam directory to launch tests on `4.5+` clusters. `4.5+` cluster needs to be configured before launching the tests against it. The files `kubeadmin-password` and `kubeconfig` which contain cluster login details should be present in the `auth` directory and it should reside in the same directory as `Makefile`. If it is not present in the auth directory, please create it. Then run `make prepare-test-cluster` to configure the `4.5+` cluster. `make prepare-test-cluster` comprises installation of sealed secrets and OpenShift GitOps operator and create sealed secrets instance.
```

Comment:
What is using the `gh` command?

Reply:
Ahh...For my next prs i kept the required binaries in the PATH. I will remove those binaries which are not needed atm.

`gh` cli is needed to login into GitHub server for cleaning up the gitops repo after a successful CI or local test run.

Reply:
removed via - 0480916